tuplex/codegen/include/BlockGeneratorVisitor.h

```diff
@@ -358,6 +362,10 @@ namespace codegen {
                 _loopBlockStack = std::deque<llvm::BasicBlock*>();
             }
 
+            if (!_loopBodyIdentifiersStack.empty()) {
```

Comment:
shouldn't _loopBodyIdentifiersStack.clear() do the same?

what about releasing the memory of the NIdentifier* members of the set?

Reply:
I'd probably write a small helper function `clearLoopBodyIdentifiers() which takes care of all that logic.

Reply:
Removed those if statements.